File: 08 -- Form Validator/Form Validator/ViewController.m
//
//  ViewController.m
//  Form Validator
//
//  Created by Joe Rocca on 3/9/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

#import "ViewController.h"
#import "FormValidator.h"

@interface ViewController () <UITextFieldDelegate>
{
    FormValidator *validator;
}

@property (weak, nonatomic) IBOutlet UITextField *nameTextField;
@property (weak, nonatomic) IBOutlet UITextField *addressTextField;
@property (weak, nonatomic) IBOutlet UITextField *cityTextField;
@property (weak, nonatomic) IBOutlet UITextField *stateTextField;
@property (weak, nonatomic) IBOutlet UITextField *zipCodeTextField;
@property (weak, nonatomic) IBOutlet UITextField *phoneNumberTextField;




@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Form Validator";
//    validator = [[FormValidator alloc] init];
    
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}




- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    BOOL rc = NO;
    
    if (textField == self.nameTextField)
    {
        rc = [validator validateName:self.nameTextField.text];
        
        
    }
    
    if (textField == self.cityTextField)
    {
        rc = [validator validateCity:self.cityTextField.text];
    }
    
    if (textField == self.zipCodeTextField)
    {
        rc = [validator validateZipCode:self.zipCodeTextField.text];
    }
    
    if (textField == self.stateTextField)
    {
        rc = [validator validateState:self.zipCodeTextField.text];
    }
    
    if(textField == self.phoneNumberTextField)
    {
        rc = [validator validatePhoneNumber:self.phoneNumberTextField.text];
    }
    
    if (textField == self.addressTextField)
    {
        rc = [validator validateAddress:self.addressTextField.text];
    }
    
    
    if (rc)
    {
        [textField resignFirstResponder];
    }
    
    return rc;
}










@end

File: 21 -- Swift Briefing/Swift Briefing copy/ViewController.swift
//
//  ViewController.swift
//  Swift Briefing copy
//
//  Created by Joe Rocca on 4/1/15.
//  Copyright (c) 2015 The Iron Yard. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITextFieldDelegate
{
    
    @IBOutlet var agentNameTextField: UITextField!
    @IBOutlet var agentPasswordTextField: UITextField!
    @IBOutlet var greetingLabel: UILabel!
    @IBOutlet var missionBriefingTextView: UITextView!
    
    
    
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        agentNameTextField.text = ""
        greetingLabel.text = "";
        missionBriefingTextView.text = "";
        
        agentNameTextField.delegate = self
        agentPasswordTextField.delegate = self
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    @IBAction func authenticateAgent(sender: UIButton)
    {
      authenticate()
    }
    

    func textFieldShouldReturn(textField: UITextField!) -> Bool
    {
        if textField == agentNameTextField
        {
        agentNameTextField.resignFirstResponder()
        agentPasswordTextField.becomeFirstResponder()
        }
        if textField == agentPasswordTextField
        {
            agentPasswordTextField.resignFirstResponder()
            
            
            if !agentPasswordTextField.text.isEmpty
            {
            authenticate()
            }
        }
        
        return true
    }
    
    
    
    
    
    func authenticate()
    {
        
        
        if !agentNameTextField.text.isEmpty && !agentPasswordTextField.text.isEmpty
        {
            var nameParts = (agentNameTextField.text.componentsSeparatedByString(" "))
            var agentName = nameParts[0]
            if nameParts.count == 2
            {
                agentName = nameParts[1]
            }
            greetingLabel.text = String(format: "Good evening, Agent %@", agentName)
            
            self.missionBriefingTextView.text = "This mission will be an arduous one, fraught with peril. You will cover much strange and unfamiliar territory. Should you choose to accept this mission, Agent \(agentName), you will certainly be disavowed, but you will be doing your country a great service. This message will self destruct in 5 seconds."
            
            
            
            self.view.backgroundColor = UIColor.greenColor()
            
            
        }
        else
        {
            self.view.backgroundColor = UIColor.redColor()
        }
        
    }
    
    
    
    
    
    
    
    
    
}

